lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1261,60 +1314,65 @@ def main():
         name=dict(required=True),
         image=dict(type='str'),
         state=dict(default="present", choices=['present', 'absent']),
-        mounts=dict(default=[], type='list'),
+        mounts=dict(default=None, type='list'),
         configs=dict(default=None, type='list'),
-        secrets=dict(default=[], type='list'),
-        networks=dict(default=[], type='list'),
+        secrets=dict(default=None, type='list'),
+        networks=dict(default=None, type='list'),
         command=dict(default=None, type='raw'),
-        args=dict(default=[], type='list'),
-        env=dict(default=[], type='list'),
+        args=dict(default=None, type='list'),
+        env=dict(default=None, type='list'),
         force_update=dict(default=False, type='bool'),
-        log_driver=dict(default="json-file", type='str'),
-        log_driver_options=dict(default={}, type='dict'),
-        publish=dict(default=[], type='list', elements='dict', options=dict(
+        log_driver=dict(default=None, type='str'),
```

Comment:
Have you checked whether any of the value options where you removed the change overrides something defined in the image? If that's the case, this can't be a Bugfix PR (similar to changing the default of `user`).

Reply:
Not passing arguments to docker which don't affect the resulting service is no problem. It's only a problem (and shouldn't be backported) if it affects the visible behavior as seen from a user (like it updates a service which hasn't been updated with the old version).